staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_whitebox_test.go

```diff
@@ -1693,3 +1693,59 @@ func TestGetCurrentResourceVersionFromStorage(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, currentPodRV, podRV, "didn't expect to see the pod's RV changed")
 }
+
+type fakeStorage struct {
+	pods []example.Pod
+	storage.Interface
+}
+
+func newObjectStorage(size int) *fakeStorage {
+	return &fakeStorage{
+		pods: func() []example.Pod {
+			items := make([]example.Pod, size, size)
+			for i := range items {
+				items[i].Namespace = "default"
+				items[i].Name = fmt.Sprintf("pod-%d", i)
+				items[i].ResourceVersion = strconv.Itoa(i)
+			}
+			return items
+		}(),
+	}
+}
+
+func (m fakeStorage) GetList(ctx context.Context, key string, opts storage.ListOptions, listObj runtime.Object) error {
+	podList := listObj.(*example.PodList)
+	podList.ListMeta = metav1.ListMeta{ResourceVersion: "12345"}
+	podList.Items = m.pods
+	return nil
+}
+func (m fakeStorage) Watch(_ context.Context, _ string, _ storage.ListOptions) (watch.Interface, error) {
+	return newDummyWatch(), nil
+}
+
+func BenchmarkCacher_GetList(b *testing.B) {
+	// create sample pods
+	cacher, _, err := newTestCacher(newObjectStorage(50000))
+	if err != nil {
+		b.Fatalf("couldn't create cacher: %v", err)
+	}
+	defer cacher.Stop()
+	// prepare result and pred
+	pred := storage.SelectionPredicate{
+		Label: labels.Everything(),
+		Field: fields.Everything(),
+	}
```

Comment:
This benchmark assumes that we actually return all objects.

The other extremely important usecase is where a bunch of objects are actually filtered out (e.g. Kubelet listing its own pods).
Can you add a second benchmark that will simulate this one (i.e. only say 50 out of those 50,000 pods are being returned)?

Reply:
Your prompt reply is much appreciated 😄!  Use-cases have been added and I provided the latest benchmark results at https://github.com/kubernetes/kubernetes/pull/116327#issuecomment-1459733316. PTAL

Reply:
Fixed, PTAL @wojtek-t